web-app/src/lib/db/services/job.service.ts

```diff
@@ -150,42 +150,9 @@ export async function startJob(
         },
       },
     });
-    await prisma.creditTransaction.update({
-      where: {
-        id: creditTransaction.id,
-      },
-      data: {
-        status: "SUCCEEDED",
-      },
-    });
-
     return job;
   } catch (error) {
-    const job = await prisma.creditTransaction.update({
-      where: {
-        id: creditTransaction.id,
-      },
-      data: {
-        status: "FAILED",
-        errorNote: error instanceof Error ? error.message : "Unknown error",
-        errorNoteKey: "Job.CreationFailed",
-      },
-      select: {
-        job: true,
-      },
-    });
-    if (job && job.job) {
-      await prisma.job.update({
-        where: {
-          id: job.job.id,
-        },
-        data: {
-          status: "FAILED",
-          errorNote: error instanceof Error ? error.message : "Unknown error",
-          errorNoteKey: "Job.CreationFailed",
-        },
-      });
-    }
+    await deleteCreditTransaction(creditTransaction.id);
```

Comment:
insteead of deleting it afterwards put everything in a prisma transaction, state updates will all succeed or all fail. No manual rollback please

Reply:
Very good idea. Thanks

d4208f0ffd30834b083b501c8fe3d27a8914aa16